cloudinit/sources/DataSourceAzure.py

```diff
@@ -1272,6 +1275,9 @@ def _negotiate(self):
 
             pubkey_info = None
             try:
+                raise KeyError(
+                    "Not using public SSH keys from IMDS"
+                )
                 public_keys = self.metadata['imds']['compute']['publicKeys']
```

Comment:
This is something of an aside (which I noticed when checking if `public_keys` being unset would be an issue), but it doesn't look like `public_keys` was used anywhere other than logging; is that a bug in the original implementation, or expected?

Reply:
That's expected that `public_keys` isn't used. That code path is just to see if the public keys are available from IMDS, otherwise if they are not than the `except` clause would need to populate a different var `pubkey_info` for further processing. Also `public_keys` is used in logging.